compiler/passes/initializerRules.cpp

```diff
@@ -2318,3 +2329,66 @@ FnSymbol* buildClassAllocator(FnSymbol* initMethod) {
 
   return fn;
 }
+
+static void buildTypeFunction(FnSymbol* fn) {
```

Comment:
I think I'm missing the whole point of buildTypeFunction().  I am surprised that it has so much AST in it
and that is pretending to "call" things.  I thought this type function would be a faux-hook that normalize
would use to break in to the C++ compiler and find/create the appropriate AggregateType for the variable?

Seems I've misunderstood the direction.  I think we'll need to resolve face to face.